### ISSUE TITLE
pg.neighbors(data_trial) from tutorial kills kernel

### ISSUE THREAD

**Issue by hoondy** [lilab-bcb/pegasus#82]
![image](https://user-images.githubusercontent.com/6531609/90288970-e5a48f00-de48-11ea-8c1f-d3b324bbb179.png)

Following the tutorial (https://pegasus.readthedocs.io/en/latest/_static/tutorials/pegasus_analysis.html), everything up until kNN clustering works but upon calling "pg.neighbors(data_trial)" crashes the kernel. I am using a conda environment with python 3.8 and the latest dev version from Github. Any help would be appreciated!

**Comment by yihming** [comment]
Hi,

I'm not able to reproduce your issue on my machine. 

I used conda environment with Python 3.8.5; knn package `hnswlib` 0.4.0; `Cython` 0.29.21; Pegasuspy master branch installed via `pip install .` from its top-level repository.

Can you try to run `import hnswlib` code in your Python interpreter, and see if this package is imported successfully?

If the issue still remains, could you please attach the output of `pip list` in your conda environment?

Thanks!

**Comment by hoondy** [comment]
I had the exact same setup as yours. I have hnswlib 0.4.0.

$ python --version
Python 3.8.5

$ pip list
Package                Version                 Location
---------------------- ----------------------- ------------------------------------------------
adjustText             0.7.3
anndata                0.7.4
annoy                  1.16.3
asciitree              0.3.3
backcall               0.2.0
bokeh                  2.1.1
certifi                2020.6.20
cffi                   1.14.1
chardet                3.0.4
click                  7.1.2
cmake                  3.18.0
colorcet               2.0.2
cycler                 0.10.0
Cython                 0.29.21
decorator              4.4.2
docopt                 0.6.2
fasteners              0.15
fbpca                  1.0
fisher                 0.1.9
fitsne                 1.1.1
forceatlas2-python     1.0
future                 0.18.2
geosketch              1.1
gprofiler-official     1.0.0
h5py                   2.10.0
harmony-pytorch        0.1.4
hnswlib                0.4.0
holoviews              1.13.3
hvplot                 0.6.0
idna                   2.10
importlib-metadata     1.7.0
intervaltree           2.1.0
ipykernel              5.3.4
ipython                7.16.1
ipython-genutils       0.2.0
jedi                   0.17.0
Jinja2                 2.11.2
joblib                 0.16.0
jupyter-client         6.1.6
jupyter-core           4.6.3
kiwisolver             1.2.0
leidenalg              0.8.1
lightgbm               2.2.1
llvmlite               0.34.0
loompy                 3.0.6
louvain                0.7.0
Markdown               3.2.2
MarkupSafe             1.1.1
matplotlib             3.3.1
mkl-fft                1.1.0
mkl-random             1.1.1
mkl-service            2.3.0
monotonic              1.5
MulticoreTSNE-modified 0.1.post2
natsort                7.0.1
numba                  0.51.0
numcodecs              0.6.4
numexpr                2.7.1
numpy                  1.19.1
numpy-groupies         0+unknown
packaging              20.4
pandas                 1.1.0
panel                  0.9.7
param                  1.9.3
parso                  0.8.0
patsy                  0.5.1
pegasusio              0.2.4.post1
pegasuspy              1.0.0rc4.dev17+g0a1b714 /sc/arion/projects/CommonMind/leed62/app/pegasus
pexpect                4.8.0
pickleshare            0.7.5
Pillow                 7.2.0
pip                    20.2.2
plotly                 4.9.0
prompt-toolkit         3.0.5
ptyprocess             0.6.0
pyarrow                1.0.0
pybind11               2.5.0
pycparser              2.20
pyct                   0.4.6
Pygments               2.6.1
pyparsing              2.4.7
python-dateutil        2.8.1
python-igraph          0.8.2
pytz                   2020.1
pyviz-comms            0.7.6
PyYAML                 5.3.1
pyzmq                  19.0.1
requests               2.24.0
retrying               1.3.3
scanorama              1.6
scikit-learn           0.23.2
scikit-misc            0.1.3
scipy                  1.5.2
scplot                 0.0.16
seaborn                0.10.1
setuptools             49.6.0.post20200814
six                    1.15.0
sortedcontainers       2.2.2
statsmodels            0.11.1
tables                 3.6.1
texttable              1.6.2
threadpoolctl          2.1.0
torch                  1.6.0
tornado                6.0.4
tqdm                   4.48.2
traitlets              4.3.3
typing-extensions      3.7.4.2
umap-learn             0.4.6
urllib3                1.25.10
wcwidth                0.2.5
wheel                  0.34.2
xlrd                   1.2.0
XlsxWriter             1.3.3
zarr                   2.4.0
zipp                   3.1.0

So the problem still persisted. I went ahead and nuked the environment and reinstalled everything from scratch. It works fine now.. Closing this issue. Thanks for looking into this and pointing me to the right direction.
